### PR TITLE
Implement filling back to a certain point

### PR DESCRIPTION
**Pull request by unbrice**
I limited refactoring to ease upstream merge.

### CODE REVIEW

File: carbonate/fill.py [hunk 1]
```diff
@@ -85,17 +85,26 @@ def fill(src, dst, tstart, tstop):
...
+    endAt is the earliest timestamp (archives are read backward).
+          if absent, we take the earliest timestamp in the archive
+    """
     header = info(dst)
     archives = header['archives']
     archives = sorted(archives, key=lambda t: t['retention'])
 
     for archive in archives:
-        fromTime = time.time() - archive['retention']
+        fromTime = max(endAt, time.time() - archive['retention'])
```
Comment (iksaif) [inline carbonate/fill.py:103]: store time.time() into now earlier ? (else time will move as we iterate)

File: tests/test_fill.py [hunk 2]
```diff
@@ -108,6 +108,38 @@ def test_fill_should_handle_gaps(self):
...
+        end = int(time.time()) + seconds_per_point
+        start = end - (points_number * seconds_per_point)
+        times = range(start, end, seconds_per_point)
+
+        complete_data = zip(times, complete)
+        self._createdb(self.db, schema, complete_data)
+        self._createdb(dst_db, schema, empty_data)
+
+        half = points_number // 2
+        fill_archives(self.db, dst_db, time.time(), time.time()-half)
```
Comment (iksaif) [inline tests/test_fill.py:165]: Can you add a test with fromTime a few seconds in the past ?

File: tests/test_fill.py [hunk 3]
```diff
@@ -108,6 +108,38 @@ def test_fill_should_handle_gaps(self):
...
+        try:
+            os.unlink(dst_db)
+        except (IOError, OSError):
+            pass
+
+        complete = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10,
+                    11, 12, 13, 14, 15, 16, 17, 18, 19, 20]
+        seconds_per_point = 1
+        points_number = len(complete)
+        schema = [(seconds_per_point, points_number)]
```
Comment (iksaif) [inline tests/test_fill.py]: can you do a test with two or three archives ?

**Comment by iksaif** [comment]
code looks good but please do a test that cover a few more cases (see comments)

**Comment by unbrice** [comment]
I added the requested test cases.

**Comment by iksaif** [comment]
LGTM

**Pull request merged**